### PR TITLE
Security fixes 2017-05

### PR DESCRIPTION
**Pull request by peterprochaska**
added new security advisories

### CODE REVIEW

File: server/oc-sa-2017-007.json [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+  "Title": "XSS in search dialogue",
+  "Timestamp": 1496219997,
+  "Risk": 1,
+  "CVSS3": {
+    "score": 2.6,
+    "vector": "CVSS:AV:N/AC:H/PR:L/UI:R/S:U/C:L/I:N/A:N"
+  },
+  "CWE": {
+    "id": 79,
+    "name": "Improper Neutralization of Input During Web Page Generation ('Cross-site Scripting')a"
```
Comment (DeepDiver1975) [inline server/oc-sa-2017-007.json:11]: )a ❓

File: server/oc-sa-2017-007.json [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+      "Operator": "<"
+    }
+  ],
+  "Description":"<p>Inadequate escaping lead to XSS vulnerability in the search module. To be exploitable an user has to write or paste malicious content into the search dialogue.</p>",
+  "ActionTaken": "Escape output",
+  "Acknowledgment":[
+    {
+      "Name":"Ahsan Khan",
+      "Reason":"Vulnerability discovery and disclosure.",
+      "HoF": true
```
Comment (DeepDiver1975) [inline server/oc-sa-2017-007.json]: HoF is not needed anymore

**Review by DeepDiver1975 — CHANGES_REQUESTED** [review]

File: server/oc-sa-2017-005.json [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+  },
+  "HackerOne": 218876,
+  "Affected":[
+    {
+      "Version":"10.0.2",
+      "CVE":"CVE-2017-9339",
+      "Operator":"<"
+    }
+  ],
+  "Description":"<p>A logical error caused disclosure of valid share tokens for public calendars. Thus granting an attacker potentially access to publicly shared calendars without knowing the share token.</p>",
```
Comment (LukasReschke) [inline server/oc-sa-2017-005.json:21]: Another friendly reminder that advisories by Nextcloud are licensed CC BY-SA 4.0. This text block is copied verbatim from https://nextcloud.com/security/advisory/?id=nc-sa-2017-011 and a Google search has shown that this word combination cannot be found somewhere else in the internet.

We expect to be credited here properly. Thank you. To comply with this licensing issue I'd like to ask you to add `"ncAdvisory": "nc-sa-2017-011",`

File: server/oc-sa-2017-007.json [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+      "CVE": "CVE-2017-9338",
+      "Operator": "<"
+    },
+    {
+      "Version": "8.2.12",
+      "CVE": "CVE-2017-9338",
+      "Operator": "<"
+    }
+  ],
+  "Description":"<p>Inadequate escaping lead to XSS vulnerability in the search module. To be exploitable an user has to write or paste malicious content into the search dialogue.</p>",
```
Comment (LukasReschke) [inline server/oc-sa-2017-007.json:35]: Another friendly reminder that advisories by Nextcloud are licensed CC BY-SA 4.0. This text block is copied verbatim from https://nextcloud.com/security/advisory/?id=nc-sa-2017-007 and a Google search has shown that this word combination cannot be found somewhere else in the internet.

We expect to be credited here properly. Thank you. To comply with this licensing issue I'd like to ask you to add `"ncAdvisory": "nc-sa-2017-007",`

**Review by LukasReschke — CHANGES_REQUESTED** [review]
It seems like you forgot to add a reference to the source as well as a notice as required by CC BY-SA 4.0. I've left a friendly reminder where I feel Nextcloud should be credited. 

In the future, I would expect that this happens without a notification from my side.

Thank you.

File: generator/generate.php [hunk 5]
```diff
@@ -149,6 +149,42 @@
...
+                $affectedVersions .= "ownCloud ". ucfirst($component). " " . $operator." ".$affected["Version"]." (".$affected["CVE"].")\n";
+            }
+            $content .= $affectedVersions;
+            if (isset($advisory['ActionTaken'])) {
+                $content .= "\nAction Taken\n";
+                $content .= "-------------\n";
+                $content .= $advisory['ActionTaken']."\n";
+            }
+
+            file_put_contents('./out/' . substr($fileinfo, 0, -5) . '.txt', $content);
```
Comment (LukasReschke) [inline generator/generate.php]: Looking at the code, this new text file doesn't contain copyright notes as per https://github.com/owncloud/security-advisories/pull/12/files#diff-064b2076f78171f7d4d71a8f1d35b1deR147 – please make sure to include this in all user-accessible formats. Thanks.

File: server/oc-sa-2017-005.json (hunk 3 above)
Comment (peterprochaska) [reply server/oc-sa-2017-005.json:21]: done. sorry

File: server/oc-sa-2017-007.json (hunk 4 above)
Comment (peterprochaska) [reply server/oc-sa-2017-007.json:35]: done

**Comment by LukasReschke** [comment]
Thanks @Peter-Prochaska.

**Comment by peterprochaska** [comment]
your are welcome @LukasReschke

**Comment by PVince81** [comment]
merge then ?

**Comment by PVince81** [comment]
old ? close ? @DeepDiver1975 @settermjd

**Comment by DeepDiver1975** [comment]
No damn idea

**Pull request closed without merging**